addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -3036,9 +3059,9 @@ var SnippetsMenu = Widget.extend({
             }
             return $target;
         };
-        globalSelector.is = function ($from) {
+        globalSelector.is = function ($from, options = {}) {
             for (var i = 0, len = selectors.length; i < len; i++) {
-                if (selectors[i].is($from)) {
+                if (options.onlyTextOptions ? $from.is(self.templateOptions[i].data.textSelector) : selectors[i].is($from)) {
```

Comment:
Mmh indeed, annoying. When I suggested this, I had in mind that data-selector would be enough but indeed, WebsiteAnimate handle both text elements and non text elements :grimacing: 

Although, I think your line here should be this:
```suggested
if (selectors[i].is($from, options)) {
```
=> Passing the options parameter and handling that in the specific `is` function (`_computeSelectorFunctions`)

Reply:
> Passing the options parameter and handling that in the specific is function (_computeSelectorFunctions)

It seems better, but I don't think there is a clean way to do it in stable without introducing a new parameter (for the `data-text-selector` value) in `_computeSelectorFunctions()`. Maybe we can do it in master FW? or add another `data = {}` parameter :grimacing: ?